.github/workflows/weighted-code-coverage.yml

```diff
@@ -168,34 +182,162 @@ jobs:
         curl -L "$GRCOV_LINK/v$GRCOV_VERSION/$GRCOV_BINARY" |
         tar xj -C $HOME/.cargo/bin
 
+    # TODO: after merge on main use the release instead of git clone and git install
```

Comment:
So you're suggesting to replace `git clone --branch tests-refactor` with just `git clone` directly here?
Because in that case we would use the `master` branch version which is very different than the `tests-refactor` one and will cause the CI checks of this pull request to fail, so that's why I noted to do it after the merge.

Reply:
Ok, let's do that after the merge then